test/Microsoft.Azure.Devices.Edge.Test.Common/certs/EdgeCertificates.cs

```diff
@@ -20,7 +16,28 @@ public class EdgeCertificates
                 new X509Certificate2(X509Certificate.CreateFromCertFile(this.TrustedCertificatesPath))
             };
 
+        private string[] GetEdgeCertFileLocation(string deviceId)
+        {
+            return new[]
+            {
+                $"certs/iot-edge-device-{deviceId}-full-chain.cert.pem",
+                $"private/iot-edge-device-{deviceId}.key.pem",
+                "certs/azure-iot-test-only.root.ca.cert.pem"
+            };
+        }
+
+        public EdgeCertificates(string deviceId, string scriptPath)
+            : base()
+        {
+            var location = this.GetEdgeCertFileLocation(deviceId);
+            var files = OsPlatform.NormalizeFiles(location, scriptPath);
+            this.CertificatePath = files[0];
+            this.KeyPath = files[1];
+            this.TrustedCertificatesPath = files[2];
+        }
+
         public EdgeCertificates(string certificatePath, string keyPath, string trustedCertsPath)
+            : base()
```

Comment:
`certificatePath` and `keyPath` should be passed to the base constructor and set there. That way you don't need to mark the base class properties as `protected set;`.

Reply:
There is an unfortunate identical constructor signature of `IdCertificates(string, string)`. I created a task for myself to refactor the X509 Cert in test framework.